Common/TilePiles/TileObject.cs

```diff
@@ -0,0 +1,107 @@
+using Microsoft.Xna.Framework.Graphics;
+using Terraria.ModLoader;
+using System.Collections.Generic;
+using System;
+
+namespace OvermorrowMod.Common.TilePiles
+{
+    public class TileObjects
+    {
+        //public static Dictionary<Type, int> TileObjectTypes;
+        //public static Dictionary<int, Texture2D> TileObjectTextures;
+        //public static Dictionary<int, string> TileObjectNames;
+        public static Dictionary<Type, string> TileObjectTypes;
+        public static Dictionary<string, Texture2D> TileObjectTextures;
+        public static Dictionary<string, string> TileObjectNames;
+
+        public static Texture2D GetTexture(string type) => TileObjectTextures[type];
+        public static string ObjectType<T>() where T : TileObject => TileObjectTypes[typeof(T)];
+
+        public static void Load()
+        {
+            TileObjectTypes = new Dictionary<Type, string>();
+            TileObjectTextures = new Dictionary<string, Texture2D>();
+            TileObjectNames = new Dictionary<string, string>();
+            TileObject.TileObjects = new Dictionary<string, TileObject>();
+        }
+
+        public static void Unload()
+        {
+            TileObjectTypes = null;
+            TileObjectTextures = null;
+            TileObjectNames = null;
+            TileObject.TileObjects = null;
+        }
+
+        public static void RegisterTileObject(Type type)
+        {
+            Type baseType = typeof(TileObject);
+            if (type.IsSubclassOf(baseType) && !type.IsAbstract && type != baseType)
+            {
+                //int id = TileObjectTypes.Count;
+                string id = type.Name;
+                TileObjectTypes.Add(type, id);
+
+                TileObject tileObject = (TileObject)Activator.CreateInstance(type);
+                tileObject.SetDefaults();
+                TileObject.TileObjects.Add(id, tileObject);
+
+                var texture = ModContent.Request<Texture2D>(tileObject.Texture ?? type.FullName.Replace('.', '/'), ReLogic.Content.AssetRequestMode.ImmediateLoad).Value;
+                TileObjectTextures.Add(id, texture);
+                TileObjectNames.Add(id, type.Name);
+            }
+        }
+    }
+
+    public class TileObject
+    {
+        public static Dictionary<string, TileObject> TileObjects;
+
+        private int _width;
+        private int _height;
+        private string _name;
+        private int _itemID;
+        private int _minStack;
+        private int _maxStack;
+
+        public int width
+        {
+            get => _width;
+            set => _width = value;
```

Comment:
This syntax for properties is a little redundant.

You can just do `pub int Width { get; set; }`, which does the same thing, pretty much.

Unless you want some special logic for your properties, just use autos.